CMakeLists.txt

```diff
@@ -155,6 +155,8 @@ find_package(UUID)
 if (UUID_FOUND)
   set(SYSLOG_NG_ENABLE_LIBUUID 1)
 endif()
+find_package(RiemannClient)
```

Comment:
In my opinion, this line does not belong to the main `CMakeLists` file, since RiemannClient is a dependency of a module, not the core of syslog-ng.

Deps should be detected and kept as close to their actual usage as possible, in order to avoid such a mess we have in configure.ac.

You probably moved the line here because you want to add `SYSLOG_NG_HAVE_RIEMANN_MICROSECONDS` to ` syslog-ng-config.h` and the variable has to be set before executing the `configure_file()` command.
I don't think this is a good idea either, `syslog-ng-config.h` should not contain details of separate modules in case of a clean CMake build. Creating multiple config.h files for modules would be a better idea IMHO (so you could actually build them separately using the syslog-ng-dev "package"), but I'm not sure we could follow this change in the autotools build..

I don't have a better idea, so this is just comment, not a request for change.

Reply:
You are right, and `SYSLOG_NG_HAVE_RIEMANN_MICROSECONDS` is module-local too, there is no need to be added to the global ones.. I'll try to find a way to keep it in the module.

Reply:
Fixed.

Reply:
Thanks.